tests/test_signals.py

```diff
@@ -0,0 +1,114 @@
+
+            version.publish(self.superuser)
+
+            self.assertEqual(env.call_count, 2)
+
+            pre_call_kwargs = env.calls[0][1]
+            post_call_kwargs = env.calls[1][1]
+
+            # pre call
+            self.assertTrue(pre_call_kwargs['token'] == post_call_kwargs['token'])
+            self.assertEqual(post_call_kwargs['operation'], constants.OPERATION_PUBLISH)
+            self.assertEqual(pre_call_kwargs['obj'], version)
+            # post call
+            self.assertTrue('token' in post_call_kwargs)
+            self.assertEqual(post_call_kwargs['operation'], constants.OPERATION_PUBLISH)
+            self.assertEqual(post_call_kwargs['obj'], version)
+
+    def test_unpublish_signals_fired(self):
+        """
+        When a version is changed to unpublished the correct signals are fired!
+        """
+        poll = factories.PollFactory()
+        version = factories.PollVersionFactory(state=constants.PUBLISHED, content__poll=poll)
+
+        with signal_tester(pre_version_operation, post_version_operation) as env:
+
+            version.unpublish(self.superuser)
+
+            self.assertEqual(env.call_count, 2)
+
+            pre_call_kwargs = env.calls[0][1]
+            post_call_kwargs = env.calls[1][1]
+
+            # pre call
+            self.assertTrue(pre_call_kwargs['token'] == post_call_kwargs['token'])
+            self.assertEqual(post_call_kwargs['operation'], constants.OPERATION_UNPUBLISH)
+            self.assertEqual(pre_call_kwargs['obj'], version)
+            # post call
+            self.assertTrue('token' in post_call_kwargs)
+            self.assertEqual(post_call_kwargs['operation'], constants.OPERATION_UNPUBLISH)
+            self.assertEqual(post_call_kwargs['obj'], version)
+
+    def test_archive_signals_fired(self):
+        """
+        When a version is changed to archived the correct signals are fired!
+        """
+        poll = factories.PollFactory()
+        version = factories.PollVersionFactory(state=constants.DRAFT, content__poll=poll)
+
+        with signal_tester(pre_version_operation, post_version_operation) as env:
+
+            version.archive(self.superuser)
+
+            self.assertEqual(env.call_count, 2)
+
+            pre_call_kwargs = env.calls[0][1]
+            post_call_kwargs = env.calls[1][1]
+
+            # pre call
+            self.assertTrue(pre_call_kwargs['token'] == post_call_kwargs['token'])
+            self.assertEqual(post_call_kwargs['operation'], constants.OPERATION_ARCHIVE)
+            self.assertEqual(pre_call_kwargs['obj'], version)
+            # post call
+            self.assertTrue('token' in post_call_kwargs)
+            self.assertEqual(post_call_kwargs['operation'], constants.OPERATION_ARCHIVE)
+            self.assertEqual(post_call_kwargs['obj'], version)
+
+    def test_draft_signals_fired(self):
+        """
+        When a version is changed to draft the correct signals are fired!
```

Comment:
The phrasing in this docstring is technically wrong as it's impossible to actually "change a version to draft". A version starts off its life as a draft, but at no point is it possible to change a version to be a draft. Which is one of the reasons I agree with Krzysztof on this one - I don't really see having a signal for draft as "adding consistency", on the contrary.

Reply:
Let's just say I have a feeling that firing a signal on Draft will be useful. Every time I have stood by something and the team push back it's later been proven that it would have been useful.